services/apps/portal/src/application/application.service.ts

```diff
@@ -69,6 +79,47 @@ export class ApplicationService {
     return this.applicationRepository.save(application);
   }
 
+  async submitToAlcs(fileNumber: string, data: ApplicationSubmitToAlcsDto) {
+    await this.update(fileNumber, data);
+    const application = await this.applicationRepository.findOneOrFail({
+      where: { fileNumber },
+      relations: { documents: true },
+    });
+
+    let submittedApp: ApplicationGrpcResponse | null = null;
+
+    try {
+      submittedApp = await lastValueFrom(
+        this.alcsApplicationService.create({
+          fileNumber: fileNumber,
+          applicant: application.applicant!,
+          localGovernmentUuid: application.localGovernmentUuid!,
+          typeCode: application.typeCode,
+          dateSubmittedToAlc: Date.now().toString(),
+          documents: application?.documents.map((d) => ({
+            type: d.type,
+            documentUuid: d.alcsDocumentUuid,
+          })),
+        }),
+      );
+    } catch (ex) {
+      this.logger.error(
+        `Portal -> ApplicationService -> submitToAlcs: failed to submit to ALCS ${fileNumber}`,
+        ex,
+      );
+
+      //TODO set failed status here?
+
+      throw new BaseServiceException(
+        `Failed to submit application: ${fileNumber}`,
+      );
+    }
+
+    //TODO set submitted status here
```

Comment:
Are you going to do another MR after this for submitted status?

Reply:
nope, this is more placeholders of when we decide to add those statuses. At this point there is not clear requirement on what should happen and how it should look like after application is submitted

Reply:
Good point!